src/GitHub.VisualStudio.UI/Helpers/ThemeDictionaryManager.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Windows;
+using System.ComponentModel;
+using Microsoft.VisualStudio.PlatformUI;
+using GitHub.VisualStudio.Helpers;
+using GitHub.Helpers;
+
+namespace GitHub.VisualStudio.UI.Helpers
+{
+    public sealed class ThemeDictionaryManager : SharedDictionaryManagerBase, IDisposable
+    {
+        static bool isInDesignMode;
+        Uri baseThemeUri;
+
+        static ThemeDictionaryManager()
+        {
+            isInDesignMode = DesignerProperties.GetIsInDesignMode(new DependencyObject());
+        }
+
+        public override Uri Source
+        {
+            get { return base.Source; }
+            set
+            {
+                InitTheme(value);
+                base.Source = GetCurrentThemeUri();
+            }
+        }
+
+        public void Dispose()
+        {
```

Comment:
I tried to sidestep this by marking the class as `sealed`. Maybe this is a bit antisocial, but it wasn't designed with extenders in mind. It kept CodeAnalysis from complaining anyway. 😉